ruleset/decoders/0540-gitlab_decoders.xml

```diff
@@ -1,106 +1,102 @@
 <!--
-  -  Gitlab v12.1 decoders.
-  -  Created by Wazuh, Inc.
-  -  Copyright (C) 2015-2021, Wazuh Inc.
-  -  This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2.
+  Copyright (C) 2015-2021, Wazuh Inc.
 -->
 
-
 <!--
-     application.log
-
-     October 06, 2014 11:56: User "Administrator" (admin@example.com) was created
-     October 06, 2014 11:56: Documentcloud created a new project "Documentcloud / Underscore"
-     October 06, 2014 11:56: Gitlab Org created a new project "Gitlab Org / Gitlab Ce"
-     October 07, 2014 11:25: User "Claudie Hodkiewicz" (nasir_stehr@olson.co.uk)  was removed
-     October 07, 2014 11:25: Project "project133" was removed
+  Decoders for:
+    GitLab: v12.
+
+  Example logs:
+    application.log
+      October 06, 2014 11:56: User "Administrator" (admin@example.com) was created
+      October 06, 2014 11:56: Documentcloud created a new project "Documentcloud / Underscore"
+      October 06, 2014 11:56: Gitlab Org created a new project "Gitlab Org / Gitlab Ce"
+      October 07, 2014 11:25: User "Claudie Hodkiewicz" (nasir_stehr@olson.co.uk)  was removed
+      October 07, 2014 11:25: Project "project133" was removed
  -->
 
-<decoder name="gitlab-12_application_log">
-    <prematch>\w+ \d+, \d\d\d\d \d\d:\d\d: </prematch>
+<decoder name="gitlab-12-application-log">
+  <prematch>\w+ \d+, \d\d\d\d \d\d:\d\d: </prematch>
 </decoder>
 
-<decoder name="gitlab_application_log_user">
-<parent>gitlab-12_application_log</parent>
-<prematch offset="after_parent">User "(\.+)" \((\.+)\)</prematch>
-<regex>(\w+ \d+, \d\d\d\d \d\d:\d\d:) User "(\.+)" \((\.+)\) was created</regex>
-<order>timestamp, new_user, e-mail</order>
+<decoder name="gitlab-application-log-user">
+  <parent>gitlab-12-application-log</parent>
+  <prematch offset="after_parent">User "(\.+)" \((\.+)\)</prematch>
```

Comment:
This prematch is redundant with the `<regex>` one line below. See as an example the following decoder where only `<regex>` is present.

Reply:
Finally they were necessary to apply the correct decoder